### PR TITLE
Adding laundry-dryer-controls cluster XML

### PR DESCRIPTION
**Pull request by eahove**
Created XML for the laundry-dryer-controls cluster and related generated code. For 0.7 ballot.

This is based on the updated spec from PR:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7766

### CODE REVIEW

**Comment by bzbarsky-apple** [comment]
How is this different from https://github.com/project-chip/connectedhomeip/pull/30098?

**Comment by eahove** [comment]
> How is this different from #30098?

This PR replaces #30098.

This PR is based on an updated version of the spec that isn't yet merged into the master branch.
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7766

**Comment by bzbarsky-apple** [comment]
@eahove Please fix CI failures and ping me on Slack when this is ready for review.

File: src/controller/data_model/controller-clusters.matter [hunk 1]
```diff
@@ -2934,6 +2934,26 @@ provisional client cluster OvenMode = 73 {
...
+            functionality available to a drying device, such as a laundry dryer. */
+client cluster LaundryDryerControls = 74 {
+  enum DrynessLevelEnum : enum8 {
+    kLow = 0;
+    kNormal = 1;
+    kExtra = 2;
+    kMax = 3;
+  }
+
+  readonly attribute optional DrynessLevelEnum supportedDrynessLevels[] = 0;
```
Comment (andy31415) [inline src/controller/data_model/controller-clusters.matter:2947]: I am looking to compare with spec in 7766. In there I do not seem to find optional conformance for either this or selectedDrynessLevel

Comment (eahove) [reply src/controller/data_model/controller-clusters.matter]: I have updated the xml with optional=false.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**